### DISCUSSION TITLE
Show the navigation bar if there is space

### DISCUSSION THREAD

**Discussion by WhyNotHugo** [pradyunsg/furo#725] [Potential Issue]
The first time I visited tox's website I was pretty confused. There didn't seem to be any navigation:

![image](https://github.com/pradyunsg/furo/assets/730811/ceb7f857-b055-4a6f-8790-f2770ad4855c)

The second time I visited the website, I noticed the tiny three lines on the upper left corner. Clicking it reveals the hidden-by-default (?) navigation bar:

![image](https://github.com/pradyunsg/furo/assets/730811/12cac494-6200-41d1-bf34-677ab9f8b53b)

At a glance, there seems to be enough space to just fit the sidebar on the left and shift the main content to the right.

I think some media queries might not be quite well tuned? The window size is exactly half of  3440x1440 (e.g.: on of the halves in a split-screen view).

**Comment by pradyunsg** [comment]
> At a glance, there seems to be enough space to just fit the sidebar on the left and shift the main content to the right.

Sadly, there isn't. :/

https://github.com/pradyunsg/furo/blob/079d829fa63091b51522c28de4e6140a9e4552a6/src/furo/assets/styles/_scaffold.sass#L348 shows the sidebar the moment there's space for one.

https://github.com/pradyunsg/furo/blob/079d829fa63091b51522c28de4e6140a9e4552a6/src/furo/assets/styles/variables/_layout.scss#L9 is the piece that leaves enough space for padding, so theoretically we could get rid of content padding or trim it but it didn't look right for me when I'd tried it. If someone has any whitespace-management alternative ideas, I'm open to them! :)

**Reply by WhyNotHugo** [reply]
The sidebar also seems to have like 30% of its width empty. Maybe it can be narrower? I'm not sure if that wouldn't mess up sites with wider titles tho.

The content padding could maybe be reduced (rather than hide the navigation entirely).

**Reply by pradyunsg** [reply]
> Maybe it can be narrower? I'm not sure if that wouldn't mess up sites with wider titles tho.

It will.

**Reply by WhyNotHugo** [reply]
The left and right margins are about 120% of the width of the sidebar.  In the screenshot above the sidebar is ~330px. The left and right sidebar are slightly over 400px.

I fail to see how "lots of empty space" is more important than "main navigation menu". _Hiding_ the entire menu by default on a desktop view which has enough space is just terrible design.

Half of the times that I land on a site using this theme I just scroll to the bottom and am totally confused as to why there is no menu, only to realise that there's a minuscule button that un-hides it. I've seen other people equally confused. The button isn't even labelled, so one has to guess what it does, and there's an almost-identical on on the other side.

**Reply by pradyunsg** [reply]
Man, I don't know what resolution we're talking about at this point. Can you please share the details of the window as reported by https://whatismyviewport.com/?

**Reply by pradyunsg** [reply]
> The window size is exactly half of 3440x1440 (e.g.: on of the halves in a split-screen view).

![Screenshot 2024-07-27 at 15 31 10](https://github.com/user-attachments/assets/3b4a15ab-9e61-4022-a090-438d37730b85)

![Screenshot 2024-07-27 at 15 31 26](https://github.com/user-attachments/assets/43a704e2-2e4e-42ad-b36d-43a15d45f5a4)

**Comment by WhyNotHugo** [comment]
https://whatismyviewport.com/ indicates:

> Your viewport size is:
>
> 1031 × 721
>
> DPR: 2.00
>
> Screen Size: 1720 × 720

I don't think that DPR and Screen Size are accurate, a fullscreen window reports:

> Your viewport size is:
>
> 2074 × 868
> 
> DPR: 2.00
> 
> Screen Size: 1720 × 720

The viewport size seems the be correct tho.

**Reply by pradyunsg** [reply]
OK, thanks for that -- I now better understand what you were saying. :)

I've pushed 05a2d63a -- let me know if that works!
